### PR TITLE
LSP skeleton [OTOT-28]

### PR DESCRIPTION
**Pull request by AlexWeinstein92**
A skeleton outline for the LSP server including workspace & text document services

### CODE REVIEW

File: server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala
Comment (reid-spencer) [inline server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala:9]: I'm not a fan of this use of a var to capture an errorCode. I presume you have a good reason for this var? Usually, when I must write a var, I also put a comment indicating why it is required.

File: server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala
Comment (reid-spencer) [inline server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala:8]: Please write a scala comment explaining what this thing is and what its base class and various services do.

File: server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPTextDocumentService.scala
Comment (reid-spencer) [inline server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPTextDocumentService.scala:10]: Shouldn't this be a class so that it can be instantiated for the return value for getTextDocumentService in RiddlLSPServer ?

**Review by reid-spencer — CHANGES_REQUESTED** [review]
Just a few things I'd like to see before we push this.

**Comment by reid-spencer** [comment]
@AlexWeinstein92 - Build and quality gate must succeed before merge.  There are compilation issues with the source code.

File: build.sbt
Comment (AlexWeinstein92) [reply build.sbt:6]: Just needed to update the IntelliJ Scala plugin

File: server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala
Comment (AlexWeinstein92) [reply server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala:9]: This resulted from translating Java code into the skeleton. Ideally we would have probably an Actor system for state management? Maybe something other than an Actor system?

File: server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala
Comment (AlexWeinstein92) [reply server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala:8]: I am thinking it will be best to do that as I build out the server. There are a lot of methods in the API for LanguageServer that could be used. We may want to have a meeting to prioritize what to do. Take a look here:
https://github.com/eclipse-lsp4j/lsp4j/blob/main/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageServer.java

**Comment by AlexWeinstein92** [comment]
@reid-spencer sonarcloud is complaining about lack of tests. For now I want to just annotate these files to say they are not to be included in coverage

**Comment by reid-spencer** [comment]
> sonarcloud is complaining about lack of tests. For now I want to just annotate these files to say they are not to be included in coverage

I'd rather you wrote the tests. TDD is a practice you should always be doing and I think you'll find how it manages to keep you on track.  The best way to get your coverage to drop is to fool yourself into thinking it is not needed. Here's a way to think of it:

You are writing a logic system.  How do you know that it is correct? The tests you write are the proof.  And, those tests will stop you from regressing through changes.  This is engineering discipline.

If you want to annotate temporarily, that's fine. But I won't approve the PR until they've all been resolved.

**Pull request merged**
